libs/providers/flagd/src/lib/service/grpc/service.ts

```diff
@@ -78,40 +75,52 @@ export class GRPCService implements Service {
   private _cacheEnabled = false;
   private _streamAlive = false;
   private _streamConnectAttempt = 0;
+  private _stream: ClientReadableStream<EventStreamResponse> | undefined = undefined;
   private _streamConnectBackoff = BASE_EVENT_STREAM_RETRY_BACKOFF_MS;
   private _maxEventStreamRetries;
   private get _cacheActive() {
     // the cache is "active" (able to be used) if the config enabled it, AND the gRPC stream is live
     return this._cacheEnabled && this._streamAlive;
   }
 
-  // default to false here - reassigned in the constructor if we actaully need to connect
-  readonly streamConnection = Promise.resolve(false);
-
-  constructor(config: Config, client?: ServiceClient, private logger?: Logger) {
+  constructor(
+    config: Config,
+    client?: ServiceClient,
+    private logger?: Logger,
+  ) {
     const { host, port, tls, socketPath } = config;
     this._maxEventStreamRetries = config.maxEventStreamRetries ?? DEFAULT_MAX_EVENT_STREAM_RETRIES;
     this._client = client
       ? client
       : new ServiceClient(
-          new GrpcTransport({
-            host: socketPath ? `unix://${socketPath}` : `${host}:${port}`,
-            channelCredentials: tls ? grpc.credentials.createSsl() : grpc.credentials.createInsecure(),
-          })
+          socketPath ? `unix://${socketPath}` : `${host}:${port}`,
+          tls ? credentials.createSsl() : credentials.createInsecure(),
         );
 
     // for now, we only need streaming if the cache is enabled (will need to be pulled out once we support events)
     if (config.cache === 'lru') {
       this._cacheEnabled = true;
       this._cache = new LRUCache({ maxSize: config.maxCacheSize || DEFAULT_MAX_CACHE_SIZE, sizeCalculation: () => 1 });
-      this.streamConnection = this.connectStream();
     }
   }
 
+  connect(
+    connectCallback: () => void,
+    changedCallback: (flagsChanged: string[]) => void,
+    disconnectCallback: () => void,
+  ): Promise<void> {
+    return this.connectStream(connectCallback, changedCallback, disconnectCallback);
+  }
+
+  disconnect(): Promise<void> {
+    this._stream?.destroy();
```

Comment:
The old generated protobuf package had no `cancel` or `destroy`. I literally couldn't figure out how to kill the stream.